scripts/travis_yml_test.py

```diff
@@ -0,0 +1,135 @@
+    # extracting test suite from patterns like --suite="accessibility"
+    hyphen_between_regex = re.compile(r'--(.+?)--')
+    suites_from_script = []
+    for script in script_raw:
+        matches = hyphen_between_regex.finditer(script)
+        for match in matches:
+            suites_from_script.append(match.group()[9:-4])
+
+    return suites_from_script
+
+
+def extract_protractor_test_suites():
+    """Extracts the test suites section from the .travis.yml file.
+
+    Returns:
+        list. A list of all test suites in protractor.conf.js file.
+    """
+    protractor_config_file = read_protractor_conf_file()
+    # Extracting suites object from protractor.conf.js.
+    suites_object_regex = re.compile(r'suites = {([^}]+)}')
+    for match in suites_object_regex.finditer(protractor_config_file):
+        suite = match.group()
+
+    comments_regex = re.compile(r'(//.*)')
+    # Removing comments of the form // from suites object.
+    suites_object = re.sub(comments_regex, '', suite)
+
+    test_suites = suites_object[10:-1]
+    words = test_suites.strip().split('\n')
+    protractor_suites = []
+    i = 0
+    # extracting tests from strings like adminPage: ['protractor_desktop/adminTabFeatures.js'],
+    unwanted_keys = ['.js', '[', ']']
+    while i < len(words):
+        key_raw = words[i].strip().split(':')[0]
+        if not any(x in key_raw for x in unwanted_keys) and len(key_raw) != 0:
+            protractor_suites.append(key_raw)
+        i += 1
+
+    return protractor_suites
+
+
+def read_protractor_conf_file():
+    """Returns the core/tests/protractor.conf.js. file."""
+    protractor_config_file = (python_utils.open_file(
+        os.path.join(
+            os.getcwd(), 'core', 'tests', 'protractor.conf.js'), 'r').read())
+    return protractor_config_file
+
+
+def read_travis_yml_file():
+    """Returns the /.travis.yml file.
+
+    Returns:
+        dict. A dict parsed from the travis.yml file.
+    """
+    travis_ci_file = python_utils.open_file(
+        os.path.join(os.getcwd(), '.travis.yml'), 'r').read()
+    travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+    return travis_ci_dict
+
+
+class TravisCIFileTests(test_utils.GenericTestBase):
+    """Test the travis ci file and protractor.conf.js have same test suites."""
+    def test_travis_and_protractor_length(self):
+
+        protractor_test_suites = extract_protractor_test_suites()
+        yaml_jobs = extract_travis_jobs_section()
+        yaml_scripts = extract_travis_script_section()
+
+        # Subtracting 4 since protractor test suites have one
+        # extra test(full: [*.js]), and three other tests that
+        # are being run by circleCi.
+        excluded_travis_tests = ['full', 'classroomPage', 'fileUploadFeatures', 'topicAndStoryEditor']
+        for excluded_test in excluded_travis_tests:
+            protractor_test_suites.remove(excluded_test)
+
+        self.assertTrue((len(yaml_jobs) == len(yaml_scripts)) and (
```

Comment:
You can drop this. The main assertion should be the one at the bottom which checks the actual strings.

However I do suggest including an assertion that the length of all lists is > 0, to ensure that these functions are actually retrieving something correctly.

Reply:
Ok Done.